imap_processing/idex/idex_l1a.py

```diff
@@ -312,9 +312,9 @@ def _set_impact_time(self, packet: space_packet_parser.parser.Packet) -> None:
               testing.
         """
         # Number of seconds since epoch (nominally the launch time)
-        seconds_since_launch = packet.data["SHCOARSE"].derived_value
+        seconds_since_launch = packet.user_data["SHCOARSE"]
```

Comment:
Basically, I don't think we should ever need to look at header/user_data. Or we should avoid it as much as possible. Just consider everything a part of the "packet" structure.
```suggestion
        seconds_since_launch = packet["SHCOARSE"]
```

Reply:
interesting...this does simplifies a lot. I am happy to do this change if others are ok with it.

Reply:
I did make changes where it was appropriate and left others as it is.